macSKK/KeyBindingSet.swift

```diff
@@ -4,19 +4,51 @@
 import AppKit
 
 struct KeyBindingSet {
-    let values: [KeyBinding.Action: [KeyBinding.Input]]
-    let dict: [KeyBinding.Input: KeyBinding.Action]
+    /**
+     * 修飾キーを除いたキー入力が同じ場合は修飾キーが多いものが前に来るように並べた配列。
+     * 入力に一番合致するキー入力を返すために最初にソートしてもっておく。
+     */
+    let sorted: [(KeyBinding.Input, KeyBinding.Action)]
 
     static let defaultKeyBindingSet = KeyBindingSet(KeyBinding.defaultKeyBindingSettings)
 
     init(_ values: [KeyBinding]) {
-        self.values = Dictionary(uniqueKeysWithValues: values.map { ($0.action, $0.inputs) })
-        self.dict = Dictionary(uniqueKeysWithValues: values.flatMap { keyValue in
+        sorted = values.flatMap { keyValue in
             keyValue.inputs.map { ($0, keyValue.action) }
+        }.sorted(by: { lts, rts in
+            if lts.0.key == rts.0.key {
+                return lts.0.modifierFlags.rawValue > rts.0.modifierFlags.rawValue
+            } else {
+                switch (lts.0.key, rts.0.key) {
+                case let (.character(l), .character(r)):
+                    return l < r
+                case let (.code(l), .code(r)):
+                    return l < r
+                case (.character, .code):
+                    return false
+                case (.code, .character):
+                    return true
```

Comment:
この2つのことから`.code < .character`っていう関係になっているってことですかね？しょうじきどういう順序で並んでいてもいいような気もしますが、

```suggestion
                // .character, .codeはどういう順序で並んでいてもいいので、いったん`.code < .character`としておく。
                case (.character, .code):
                    return false
                case (.code, .character):
                    return true
```

いちおうこういう ☝️ コメントを残しておきますかね？ 🤔

Reply:
ですね > `どういう順序で並んでいてもいいような気もします`

ただ、意図がコードから読み取りづらいのでコメント残しておくことにします。